### PR TITLE
Add Bluetooth serial port explainer.

### PR DESCRIPTION
**Pull request by cmumford**

### CODE REVIEW

File: EXPLAINER_BLUETOOTH.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+   created a device node for that port.
+3. Communicate with a non-Serial Port service that exposes a serial interface.
+   See [Non-standard Service Class ID’s](#non-standard-service-class-ids) below.
+
+## Why not enhance Web Bluetooth?
+
+The intent is to avoid the tendency to backfill Bluetooth Classic support into
+[Web Bluetooth](https://webbluetoothcg.github.io/web-bluetooth/). If support for
+one profile were to land in Web Bluetooth, this might entice developers to
+request the addition of other classic profiles to Web Bluetooth. Support for
```
Comment (reillyeon) [inline EXPLAINER_BLUETOOTH.md:58]:
```suggestion
one profile were to land in Web Bluetooth, developers might
request the addition of other classic profiles to Web Bluetooth. Support for
```

File: EXPLAINER_BLUETOOTH.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+Bluetooth serial interfaces is seen as a special case as It has come numerous
+times in the form of developer requests needing basic serial port support with
```
Comment (reillyeon) [inline EXPLAINER_BLUETOOTH.md:60]:
```suggestion
Bluetooth serial interfaces is seen as a special case as it is so frequently
requested by developers needing basic serial port support with
```

File: EXPLAINER_BLUETOOTH.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+and macOS a pair of filesystem nodes in /dev (i.e. `/dev/cu.*` and
+`/dev/tty.*`).
+
+This proposal will allow support for Bluetooth services that implement the
+Serial Port Profile even if the operating system does not create device nodes
+for Bluetooth serial ports.
+
+## Non-standard Service Class ID’s
+
+Devices supporting a Serial Port Profile
```
Comment (reillyeon) [inline EXPLAINER_BLUETOOTH.md:80]:
```suggestion
Devices supporting the Serial Port Profile
```

File: EXPLAINER_BLUETOOTH.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+## Potential specification changes
+
+### Port Selection
+
+[`Serial.requestPort()`](https://developer.mozilla.org/en-US/docs/Web/API/Serial/requestPort),
+with no filters parameter, will pass through:
+
+1. Non-Bluetooth serial ports (wired, virtual, etc.)
+2. Mapped Bluetooth serial ports.
```
Comment (reillyeon) [inline EXPLAINER_BLUETOOTH.md:104]: It's not clear at this point what "mapped" means. I think you mean bonded Bluetooth devices where the operating system has automatically created device nodes for the Serial Port service.

File: EXPLAINER_BLUETOOTH.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+3. Any unmapped serial port provided by a standard Bluetooth SerialPort service.
+   These will appear on operating systems without automatic port mapping.
+
+### Service Class ID Filtering
+
+Serial ports for services other than the SerialPort service (UUID: `0x1101`)
+ can be requested, but will be omitted unless explicitly filtered for.
```
Comment (reillyeon) [inline EXPLAINER_BLUETOOTH.md:111]:
```suggestion
can be requested, but will be omitted unless explicitly filtered for.
```

File: EXPLAINER_BLUETOOTH.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+1. `usbVendorId`
+2. `usbProductId`
+
+To provide the ability to filter Bluetooth serial ports filters will need to
+support filtering for a Bluetooth service identified by a service class ID. This
+will be implemented by adding the following SerialPortFilter member:
+
+1. `bluetoothServiceClassId`
+
+Browser vendors may provide UI for the user to select certain device types such
```
Comment (reillyeon) [inline EXPLAINER_BLUETOOTH.md:130]:
```suggestion
Implementations may provide UI for the user to select certain device types such
```

File: EXPLAINER_BLUETOOTH.md [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+Currently
+[`SerialPort.getInfo()`](https://developer.mozilla.org/en-US/docs/Web/API/SerialPort/getInfo)
+returns a [`SerialPortInfo`](https://wicg.github.io/serial/#serialportinfo-dictionary)
+dictionary containing usbVendorId and usbProductId.
+
+SerialPortInfo will be extended by adding the following member:
+
+1. `bluetoothServiceClassId` (string) – The Bluetooth service class UUID. This may
+   be either a 16 or 128 bit UUID.
```
Comment (reillyeon) [inline EXPLAINER_BLUETOOTH.md:143]: We should probably normalize this to the 128-bit UUID form as Web Bluetooth does.

**Review by reillyeon — APPROVED** [review]
Looks good, with a couple minor comments.

File: EXPLAINER_BLUETOOTH.md (hunk 4 above)
Comment (cmumford) [reply EXPLAINER_BLUETOOTH.md:104]: I discuss this tangentially in "Support for unmapped serial ports" above (currently line 66). I'm using "mapped" to describe the creation of one (or more) serial device nodes existing for a single Bluetooth serial port. They can be automatically created (which Windows and macOS do), or manually (like using `rfcomm` on Linux). Do you think "map" is a good term for this, or is there a better one?

File: EXPLAINER_BLUETOOTH.md (hunk 5 above)
Comment (cmumford) [reply EXPLAINER_BLUETOOTH.md:111]: done.

File: EXPLAINER_BLUETOOTH.md (hunk 6 above)
Comment (cmumford) [reply EXPLAINER_BLUETOOTH.md:130]: done

File: EXPLAINER_BLUETOOTH.md (hunk 7 above)
Comment (cmumford) [reply EXPLAINER_BLUETOOTH.md:143]: done.

File: EXPLAINER_BLUETOOTH.md (hunk 3 above)
Comment (cmumford) [reply EXPLAINER_BLUETOOTH.md:80]: done.

File: EXPLAINER_BLUETOOTH.md (hunk 2 above)
Comment (cmumford) [reply EXPLAINER_BLUETOOTH.md:60]: done.

File: EXPLAINER_BLUETOOTH.md (hunk 1 above)
Comment (cmumford) [reply EXPLAINER_BLUETOOTH.md:58]: done.

File: EXPLAINER_BLUETOOTH.md (hunk 4 above)
Comment (reillyeon) [reply EXPLAINER_BLUETOOTH.md:104]: I think I withdraw the question because I just didn't read the section heading above carefully enough.

**Comment by cmumford** [comment]
Is there anything else I need to do for this to be merged?

**Pull request merged**